src/Heap.js

```diff
@@ -51,34 +77,61 @@ export default class Heap {
     }
 
     buildMinHeap() { 
-        const lastParent = Math.floor((this.heap.length - 2) / 2);
+        const lastParent = Math.floor((this.heap.length - 1) / 2);
 
         for (let i = lastParent; i >= 0; i--) {
             this.minHeapify(i);
         }
     }
 
     checkHeap(position) {
-        while (position> 0) {
-            position = Math.floor((position - 2) / 2);
+        while (position > 0) {
+            position = Math.floor((position - 1) / 2);
             this.minHeapify(position);
         }
     }
 
     insert(node, key) {
         this.heap.push({node, key});
         this._nodeIndices.set(node, this.heap.length - 1);
-        this.checkHeap(this.heap.length - 1);
+        this.filterUp(this.heap.length - 1);
     }
 
     compare(a, b) {
         if (a && b) {
```

Comment:
is there ever a case where `!a` or `!b` is an actual case we should ignore?

reminder that you also wanted this to work on straight integers for A*, so..... this breaks if a key is 0

but, furthermore... when will these ever not be a number or array?

Reply:
🐇